src/shims/unix/linux/epoll.rs

```diff
@@ -409,11 +409,12 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         )?;
         let timeout = this.read_scalar(timeout)?.to_i32()?;
 
-        if epfd <= 0 {
+        if epfd <= 0 || maxevents <= 0 {
```

Comment:
The `maxevents.try_into().unwrap()` above needs to be moved down after this check, it would ICE on a negative `maxevents`.